cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -1160,6 +1160,13 @@ private Future<MapAndStatus<ConfigMap, KafkaRebalanceStatus>> onReady(Reconcilia
                                 new NoSuchResourceException("Kafka resource '" + clusterName
                                         + "' identified by label '" + Labels.STRIMZI_CLUSTER_LABEL
                                         + "' does not exist in namespace " + clusterNamespace + ".")).mapEmpty();
+                    } else if (kafka.getStatus().getConditions().stream().noneMatch(condition -> condition.getType().equals("Ready") && condition.getStatus().equals("True"))) {
+                        String errorString = "Kafka cluster is being deployed";
+                        LOGGER.warnCr(reconciliation, errorString);
+                        KafkaRebalanceStatus status = new KafkaRebalanceStatus();
+                        status.addCondition(StatusUtils.buildWarningCondition(CruiseControlIssues.kafkaClusterNotReady.getReason(), errorString));
+                        return updateStatus(reconciliation, kafkaRebalance, status,
+                                new RuntimeException("Secret " + clusterNamespace + "/my-cluster-cruise-control-certs does not exist")).mapEmpty();
```

Comment:
Shouldn't the RuntimeException message here be related to the Kafka cluster not being ready instead of a secret missing?

Reply:
Actually we are getting two conditions in the status. This Runtime Exception is an exception condition which moves the rebalance resource to `NotReady` and states the erro due to which the resource went to `NotReady` state. The other condition is `kafkaClusterNotReady` which is basically a warning condition stating that the Kafka resource is not ready yet

Reply:
Sorry, I don't get the explanation. I'm with Kyle here. You check that the cluster is not ready. Nothing about the secret. So that should be the error here.

Reply:
Do we need both conditions? The secret  "does not exist" runtime error happens if the KafkaRebalanceOperator attempts to connect to a Kafka cluster that is not ready right? If the new logic added prevents that and knows when the Kafka cluster is not ready, shouldn't we just have that in the status, the the Kafka cluster is not ready?

Reply:
The problem here is that the new check is much wider then just checking the secret. So it does not know anything about the secrets existence. It might be not ready because some worker node went down and one ZooKeeper pod is pending for example.